src/expr.h

```diff
@@ -80,7 +80,7 @@ class Expr : public ASTNode {
         constant of the given type, this method should return the
         corresponding llvm::Constant value.  Otherwise it should return
         NULL. */
-    virtual llvm::Constant *GetConstant(const Type *type) const;
+    virtual std::pair<llvm::Constant *, bool> GetConstant(const Type *type) const;
```

Comment:
You need comments explaining the new function interface and how it's supposed to be used.

Reply:
For the second argument I suggest using `enum class` consisting of two values - this way semantics is much more explicit and it's harder to misinterpret the result.

Reply:
The problem with making it an 'enum class' is ORing values in case of recursive calls. Everytime GetConstant is called, it might recursively call GetConstant on part of that expresion and we need to ORvalue in current GetConstant with the called GetConstant. If its an enum, this becomes a little clunky. Let me know if I understand this correctly

Reply:
Ok, let's keep it a bool.